src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderParser.cs

```diff
@@ -1161,6 +1161,11 @@ private bool FetchNextField(ref ScanInfo scan, ReadOnlySpan<char> span)
                                 scan.QuotingError = true;
                                 break;
                             }
+
+                            // The logic below allow us to scape quotes (") inside quoted
+                            // fields by using doublo quotes (""). I.e. when the loader
+                            // encounters "" inside a quoted field, it will output only one "
+                            // and continue parsing the rest of the field.
                             if (span[ichCur] == '"')
```

Comment:
Might want to expose this character as a parameter. The other common choice is backslash escaping.

Reply:
To implement the feature you're suggesting [on this comment](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425735487) I do believe that we only need to change the code you've pointed to in here, so it's kind of straightforward.

Right now I am going to focus on making sure this PR works well with AutoML/Modelbuilder, that TextSaver works with new lines and that we handle correctly the case of having a badly formatted input file. If I can tackle this with enough time before the next release, then I will try to implement this suggestion.

---
In reply to: [425777440](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425777440) [](ancestors = 425777440)

Reply:
Change in plans, I won't address surfacing the new readMultiline option to AutoML/ModelBuilder on this PR. So now I will actually focus on the other things, including implementing this escapechar option. See [https://github.com/dotnet/machinelearning/pull/5125#discussion_r425981453](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425981453)

---
In reply to: [425777440](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425777440) [](ancestors = 425777440)

Reply:
Note to myself: I've just realized that to implement the escapechar feature, then I'll also need to change the logic in the MultilineReader, because if the escapechar isn't `"` then checking for even number of quotes won't be enough.

---
In reply to: [425777440](https://github.com/dotnet/machinelearning/pull/5125#discussion_r425777440) [](ancestors = 425777440)

Reply:
As discussed offline, I'll address this in an upcoming PR.

---
In reply to: [426128704](https://github.com/dotnet/machinelearning/pull/5125#discussion_r426128704) [](ancestors = 426128704,425777440)